osc/conf.py

```diff
@@ -588,17 +588,20 @@ def list_ssh_dir_keys(self):
             sshdir = os.path.expanduser('~/.ssh')
             keys_in_home_ssh = {}
             for keyfile in os.listdir(sshdir):
-                if keyfile.endswith(".pub"):
-                    continue
                 keyfile_path = os.path.join(sshdir, keyfile)
-                if not self.is_ssh_private_keyfile(keyfile_path):
+                if (not os.path.isfile(keyfile_path) or
+                        keyfile.startswith("authorized_keys") or
+                        keyfile == "config" or
+                        keyfile.startswith("agent-") or
+                        keyfile.startswith("known_hosts")):
```

Comment:
I'm not sure this logic is sound. In cases where a user has multiple key pairs and only one of them is supposed to be used for `osc`, this may choke on other private keys, no?
It should probably be `if (not self.is_ssh_private_keyfile) and (not self.is_ssh_public_keyfile)`.

Reply:
Right, the logic doesn't work with multiple keys. But that was the also the case previously, unless I'm mistaken (the code would have picked up every private key file). If you have multiple keys, you need to put `sshkey=...` in your `.oscrc`.

> It should probably be if (not self.is_ssh_private_keyfile) and (not self.is_ssh_public_keyfile).

I was too lazy to code up `is_ssh_public_keyfile()` :grin:, I believe that's non-trivial. I was relying on the fact that `ssh-keygen -lf` would fail for anything that's not either a private or a public key file.